packages/stratimikos/test/CMakeLists.txt

```diff
@@ -244,8 +244,9 @@ IF (${PROJECT_NAME}_ENABLE_Tpetra AND ${PACKAGE_NAME}_ENABLE_ThyraTpetraAdapters
     EXCLUDE_IF_NOT_TRUE ${PROJECT_NAME}_ENABLE_FLOAT
     )
 
+  ASSERT_DEFINED(Belos_SOURCE_DIR)
+
   TRIBITS_COPY_FILES_TO_BINARY_DIR(CopyTpetraTestMtxFiles
-    ASSERT_DEFINED(Belos_SOURCE_DIR)
```

Comment:
It is shocking that CMake even allowed a macro call in the middle of another macro call.

Reply:
Actually, it is not shocking because CMake treats all of the arguments to a function or macro as raw strings so it sees this as just the string `"ASSERT_DEFINED(Belos_SOURCE_DIR)"` argument.  But the implementation of `tribits_copy_files_to_binary_dir()` at:

https://github.com/TriBITSPub/TriBITS/blob/d1cd73cb2daa80baa4d712c102c3712259b70210/tribits/core/package_arch/TribitsCopyFilesToBinaryDir.cmake#L143-L155

should have caught that and reported this as a unparsed error.

@cgcgcg, what output did you see that caused you to fix this?  I am not seeing any CMake configure warnings before this PR and after this was inserted in the commit 7de66b2dfb62bbc06dd7b31f1ecc576d697078a2.

Reply:
This is what I saw:
```
CMake Warning at cmake/tribits/core/utils/MessageWrapper.cmake:47 (message):
  Arguments passed in unrecognized.  PARSE_UNPARSED_ARGUMENTS =
  'ASSERT_DEFINED;(;Belos_SOURCE_DIR;)'
Call Stack (most recent call first):
  cmake/tribits/core/utils/TribitsParseArgumentsHelpers.cmake:46 (message_wrapper)
  cmake/tribits/core/package_arch/TribitsCopyFilesToBinaryDir.cmake:155 (tribits_check_for_unparsed_arguments)
  packages/stratimikos/test/CMakeLists.txt:247 (TRIBITS_COPY_FILES_TO_BINARY_DIR)
```